### ISSUE TITLE
Retry requests to executor state

### ISSUE THREAD

**Issue by wyegelwel** [twosigma/Cook#155]
Currently requests to get executor state are not retried so users can sometimes not get an output_url back even if the host is up. Retrying the request would help mitigate the risk. 
https://github.com/twosigma/Cook/blob/master/scheduler/src/cook/mesos/api.clj#L280

**Comment by DaoWen** [comment]
@shamsimam - is this fixed when using the cook executor?

**Comment by shamsimam** [comment]
Yes, for tasks launched to run with the Cook executor. For tasks that are run using the Mesos executor, I suspect that this error is still there but we probably care less about them?

**Comment by DaoWen** [comment]
@wyegelwel - What do you think about closing this? (The resolution being to use the Cook Executor.)
